include/shell/shell.h

```diff
@@ -593,6 +594,16 @@ void shell_print_stream(const void *user_ctx, const char *data,
  */
 int shell_execute_cmd(const struct shell *shell, const char *cmd);
 
+/** @brief Defines shell_fprintf behavior on CR and NL characters.
+ *
+ *
+ * @param[in] shell	Pointer to the shell instance.
+ * @param[in] flag	Requested behavior.
+ *
+ * @returns Returns 0 on success or error code.
+ */
+int shell_set_flags(const struct shell *shell, enum shell_flag flag);
```

Comment:
If you allowing to set multiple flags, please do not use enum to pass them.
If not, use "flag" instead of "flags" in function name.